### ISSUE TITLE
TestBug323833 fails on Mac since I20240729-1820

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform#1504]
Both mac64 and macM:
https://download.eclipse.org/eclipse/downloads/drops4/I20240813-0100/testresults/html/org.eclipse.core.tests.resources_ep433I-unit-mac64-java17_macosx.cocoa.x86_64_17.html
https://download.eclipse.org/eclipse/downloads/drops4/I20240813-0100/testresults/html/org.eclipse.core.tests.resources_ep433I-unit-macM1-java17_macosx.cocoa.aarch64_17.html

```
org.opentest4j.AssertionFailedError: expected: <true> but was: <false>
at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
at org.junit.jupiter.api.AssertTrue.failNotTrue(AssertTrue.java:63)
at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:36)
at org.junit.jupiter.api.AssertTrue.assertTrue(AssertTrue.java:31)
at org.junit.jupiter.api.Assertions.assertTrue(Assertions.java:183)
at org.eclipse.core.tests.resources.session.TestBug323833.test1(TestBug323833.java:85)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.eclipse.core.tests.harness.session.SessionTestExtensionRemote.skipIfExecuteInHost(SessionTestExtensionRemote.java:61)
at org.eclipse.core.tests.harness.session.SessionTestExtensionRemote.interceptTestMethod(SessionTestExtensionRemote.java:52)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

@HannesWell all commits in that timeframe come from you and are related to the filesystem which is tested by the test. Please investigate. It would be helpful if you check the next i-build yourself after every commit.

**Comment by HannesWell** [comment]
> @HannesWell all commits in that timeframe come from you and are related to the filesystem which is tested by the test. Please investigate. It would be helpful if you check the next i-build yourself after every commit.

From looking at the changes in that time-frame (https://github.com/eclipse-platform/eclipse.platform/compare/I20240722-0320...I20240729-1820) my first guess is that `Files.copy()` copies File-attributes differently than the Eclipse Native(File)Handler on Mac. So https://github.com/eclipse-platform/eclipse.platform/pull/1471 is maybe a cause.
If that's right https://github.com/eclipse-platform/eclipse.platform/pull/1502, will fix this for small files and I created https://github.com/eclipse-platform/eclipse.platform/pull/1512 to do the same for large files.

But since I don't have access to a Mac I cannot verify this by myself. What would help the most if the I-build was consistent with verification builds. Then one would have not to check the I-builds everytime.

**Comment by jukzi** [comment]
> [I20240722-0320...I20240729-1820](https://github.com/eclipse-platform/eclipse.platform/compare/I20240722-0320...I20240729-1820)
That's a nice query!

**Comment by HannesWell** [comment]
> If that's right #1502, will fix this for small files and I created #1512 to do the same for large files.

In the latest I-Build this error has vanished, so I assume that assumption was correct.
